streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -281,48 +275,63 @@
     <V1, V2> KStream<K, V2> leftJoin(KTable<K, V1> table, ValueJoiner<V, V1, V2> joiner);
 
     /**
-     * Aggregate values of this stream by key on a window basis.
+     * Combine values of this stream by key on a window basis into a new instance of windowed {@link KTable}.
```

Comment:
IMHO because the behavior of returning a windowed KTable for reduceByKey might be unfamiliar to many users, I'd extend the documentation of this method a bit -- a 1-liner is not sufficient if you ask me.